### PR TITLE
docs(chore): Update Link docs

### PR DESCRIPTION
**Pull request by joebuono**
#### Description of changes

This PR updates the Link docs page. 

**Changes/fixes:**
- Include Global and Local styling examples which demonstrate how to style each of the pseudo-classes (theme example is already included in [this PR](https://github.com/aws-amplify/amplify-ui/pull/2107))
- For the Routing Libraries example, use this code: https://github.com/aws-amplify/amplify-ui/issues/705#issuecomment-1033236546
- Toggling `isExternal` in the main demo shows `undefined` as the children
- For the External Links description, mention that `isExternal` will cause the link to be opened up in a new tab. Also mention `rel=“noopenner noreferrer”`
- Links should all have an `href`, even if it’s just ui.docs.amplify.aws
- External Links and Routing Libraries should be nested under Usage
- Update main demo
- Various nits

#### Description of how you validated changes

Ran the docs locally.

#### Checklist

- [X] PR description included
- [X] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by hbuchel — APPROVED** [review]
Pulled down and checked  it out locally, lgtm!

File: docs/src/pages/[platform]/components/link/LinkPropControls.tsx [hunk 1]
```diff
@@ -20,19 +27,29 @@ export const LinkPropControls: LinkPropControlsInterface = ({
...
   setTextDecoration,
+  children,
+  setChildren,
 }) => {
   return (
     <Flex direction="column">
+      <TextField
+        value={children as string}
+        onChange={(event) => setChildren(event.target.value)}
+        label="body"
```
Comment (reesscot) [inline docs/src/pages/[platform]/components/link/LinkPropControls.tsx:38]: Why label this as `body` when it's demonstrating the `children` prop?

File: docs/src/pages/[platform]/components/link/LinkPropControls.tsx [hunk 2]
```diff
@@ -20,19 +27,29 @@ export const LinkPropControls: LinkPropControlsInterface = ({
...
       />
-      <TextField
+      <SelectField
         value={textDecoration as string}
-        onChange={(e) => setTextDecoration(e.target.value)}
+        onChange={(event) => setTextDecoration(event.target.value)}
         label="textDecoration"
-      />
+      >
+        <option value="none">none</option>
```
Comment (reesscot) [inline docs/src/pages/[platform]/components/link/LinkPropControls.tsx:50]: non-blocking (could be follow up): Should we include a few more options? https://developer.mozilla.org/en-US/docs/Web/CSS/text-decoration

File: docs/src/pages/[platform]/components/link/LinkPropControls.tsx (hunk 1 above)
Comment (joebuono) [reply docs/src/pages/[platform]/components/link/LinkPropControls.tsx:38]: I was following the pattern from the [Alert docs](https://ui.docs.amplify.aws/react/components/alert). But I agree `children` makes more sense, I'll change it

File: docs/src/pages/[platform]/components/link/LinkPropControls.tsx [hunk 3]
```diff
@@ -20,19 +27,35 @@ export const LinkPropControls: LinkPropControlsInterface = ({
...
+        <option value="underline">underline</option>
+        <option value="underline overline #FF3028">
+          underline overline #FF3028
+        </option>
+        <option value="underline dotted">underline dotted</option>
+        <option value="underline dotted red">underline dotted red</option>
+        <option value="green wavy underline">green wavy underline</option>
```
Comment (hbuchel) [inline docs/src/pages/[platform]/components/link/LinkPropControls.tsx:57]: I did not know wavy was an option lol

**Review by hbuchel — APPROVED** [review]

**Review by reesscot — APPROVED** [review]
LGTM, one non-blocking nit

File: docs/src/pages/[platform]/components/link/demo.tsx [hunk 4]
```diff
@@ -9,8 +9,9 @@ import { demoState } from '@/utils/demoState';
 const propsToCode = (props: LinkProps) => {
   return (
     '<Link' +
+    '\n href="https://ui.docs.amplify.aws/react/components/link"' +
     (props.color ? `\n color="${props.color}"` : '') +
-    (props.textDecoration
+    (props.textDecoration !== 'none'
```
[the remark this answers is not in the document]
Comment (joebuono) [reply docs/src/pages/[platform]/components/link/demo.tsx:14]: Good point. I'll merge for now and fix it later

**Pull request merged**